Sources/Sentry/SentryFramesTracker.m

```diff
@@ -106,26 +125,45 @@ - (void)displayLinkCallback
     CFTimeInterval slowFrameThreshold = 1 / (actualFramesPerSecond - 1);
 
     CFTimeInterval frameDuration = lastFrameTimestamp - self.previousFrameTimestamp;
-    self.previousFrameTimestamp = lastFrameTimestamp;
 
     if (frameDuration > slowFrameThreshold && frameDuration <= SentryFrozenFrameThreshold) {
         atomic_fetch_add_explicit(&_slowFrames, 1, SentryFramesMemoryOrder);
-    }
-
-    if (frameDuration > SentryFrozenFrameThreshold) {
+#    if SENTRY_TARGET_PROFILING_SUPPORTED
+        [self recordTimestampStart:@(self.previousFrameTimestamp) end:@(lastFrameTimestamp)];
+#    endif // SENTRY_TARGET_PROFILING_SUPPORTED
+    } else if (frameDuration > SentryFrozenFrameThreshold) {
         atomic_fetch_add_explicit(&_frozenFrames, 1, SentryFramesMemoryOrder);
+#    if SENTRY_TARGET_PROFILING_SUPPORTED
+        [self recordTimestampStart:@(self.previousFrameTimestamp) end:@(lastFrameTimestamp)];
+#    endif // SENTRY_TARGET_PROFILING_SUPPORTED
```

Comment:
`m`: So we only record the slow and frozen frames? If yes, I think we can drastically reduce the SentryNumberOfFrameTimestampsToRetain`. Can we also document the decision here if so?